nrf_security/src/mbedtls/mbedtls_heap.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <init.h>
+#include <app_memory/app_memdomain.h>
+
+#include "mbedtls/memory_buffer_alloc.h"
+
+#if !defined(CONFIG_MBEDTLS_HEAP_SIZE) || CONFIG_MBEDTLS_HEAP_SIZE == 0
+#error "CONFIG_MBEDTLS_HEAP_SIZE must be specified and greater than 0"
+#endif
+
+static unsigned char mbedtls_heap[CONFIG_MBEDTLS_HEAP_SIZE];
+
+static int mbedtls_heap_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	mbedtls_memory_buffer_alloc_init(mbedtls_heap, sizeof(mbedtls_heap));
+
+	return 0;
+}
+
+/* Entropy cc310 is initialized with CONFIG_KERNEL_INIT_PRIORITY_DEVICE and the
```

Comment:
hw_cc310 is the thing that must be initialized before

Reply:
but entropy must be init before heap, correct ?
I prefer to only state first dependency here, and then we should consider to state the entropy -> hw dependency in the entropy driver.

Reply:
entropy is only dependent on hw_cc310 (because it requires mutex-usage). So the chain is hw_cc310->heap or hw_cc310->entropy_cc310->heap (dependent of the presence of entropy in your configuraiton)

Reply:
So entropy and heap could happen on same level, actually... They are not dependent on each other (since the entropy_cc310 driver only requires mutex-support, and the same is the thing for heap)

Reply:
then I will change priority to match entropy priority as that will ensure mbedTLS is ready before app init prio.
No reason to keep mbed TLS heap on application init priority.